### ISSUE TITLE
File Uploading is extremely slow

### ISSUE THREAD

**Issue by himabindugit** [transmission/transmission#4438]
### What is the issue?

In summary, transferring data from EC2 instance to an on-premises system. The data is compressed in a tarball and then broken into smaller chunks to be transferred via BitTorrent client. We can successfully transfer a 6.31 GB file this way within 40 minutes (~24Mb/s at 35min), but with a different file of equal size, this transfer is stalling (stuck for more than an hour after only 773kb transferred). 
We tried benchmark the data transfer speed from an EC2 instance to the server manually using `iperf` (with out Bittorrent), when sending data from the EC2 instance to the server we are getting a transfer rate of 29.7 Mbits/sec which is a lot faster. This rules out any issues with the network. what could be the cause of the slow transfer? We enabled transmissionrpc logger to see more logs, Is there any specific log we should be logging? Is there anything else we can try?

### Which application of Transmission?

transmission-daemon

### Which version of Transmission?

Python library [transmission-rpc](https://pypi.org/project/transmissionrpc/)

**Comment by mikedld** [comment]
Please specify the version of Transmission. We're not providing support for 3rd-party libraries here, and those libraries aren't the ones doing the actual data transfer so they're unlikely to matter.

**Comment by himabindugit** [comment]
Transmission 1.31 - 2.82

**Comment by mikedld** [comment]
v2.82 is 6+ years old. Unless you're willing to try out the newer version (ideally, 4.x), I'm not sure we'll be able to help.

**Comment by himabindugit** [comment]
Ok, Thank you

**Comment by twerq** [comment]
In my case upload speed won't get more than 70KB/s. And usually I see 2 - 10 KB/s. This is the first time I've encountered this kind of behavior with transmission 3.0 from official Linux Mint 21.1 repositories. And today I've dropped 3.0 version. I've got the sources of 4.0 beta 3 and have built. It looks running well, but gives just the same upload speeds as I transmission 3.0 show.
Notice: I have windows in a dual boot on my machine and never saw so low upload in utorrent app
